service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/SpecimenMapper.java

```diff
@@ -28,69 +37,214 @@
+            pertinentInformation = StringUtils.EMPTY;
+        }
+
+        public void fastingStatus(CodeableConcept fastingStatus) {
+            CodeableConceptMappingUtils.extractTextOrCoding(fastingStatus)
+                .map(fastingStatusValue -> newLine(withSpace(FASTING_STATUS, fastingStatusValue), pertinentInformation))
+                .ifPresent(text -> pertinentInformation = text);
+        }
+
+        public void fastingDuration(Duration fastingDuration) {
+            List<String> fastingDurationElements = List.of(
+                FASTING_DURATION,
+                Objects.toString(fastingDuration.getValue(), StringUtils.EMPTY),
+                fastingDuration.getUnit()
             );
+
+            pertinentInformation = newLine(withSpace(fastingDurationElements), pertinentInformation);
         }
 
-        if (specimen.hasReceivedTime()) {
-            return String.format(
-                EFFECTIVE_TIME_TEMPLATE,
-                DateFormatUtil.toHl7Format(specimen.getReceivedTimeElement())
+        public void quantity(SimpleQuantity quantity) {
+            List<String> quantityElements = List.of(
+                QUANTITY,
+                Objects.toString(quantity.getValue(), StringUtils.EMPTY),
+                quantity.getUnit()
             );
+
+            pertinentInformation = newLine(withSpace(quantityElements), pertinentInformation);
+        }
+
+        public void collectionSite(CodeableConcept collectionSite) {
+            CodeableConceptMappingUtils.extractTextOrCoding(collectionSite)
+                .map(collectionSiteValue -> newLine(withSpace(COLLECTION_SITE, collectionSiteValue), pertinentInformation))
+                .ifPresent(text -> pertinentInformation = text);
+        }
+
+        public void note(String note) {
+            pertinentInformation = newLine(note, pertinentInformation);
+        }
+
+        public Optional<String> build(DateTimeType date) {
+            if (StringUtils.isNotBlank(pertinentInformation)) {
+                return Optional.of(newLine(withSpace(COMMENT_PREFIX, DateFormatUtil.toTextFormat(date)), pertinentInformation));
+            }
+
+            return Optional.empty();
         }
 
-        return StringUtils.EMPTY;
+        public Optional<String> build() {
+            if (StringUtils.isNotBlank(pertinentInformation)) {
+                return Optional.of(newLine(COMMENT_PREFIX, pertinentInformation));
+            }
+
+            return Optional.empty();
+        }
+
+        private String newLine(Object... values) {
+            return StringUtils.joinWith(StringUtils.LF, values);
+        }
+
+        private String withSpace(Object... values) {
+            return StringUtils.join(values, StringUtils.SPACE);
+        }
+
+        private String withSpace(List<String> values) {
+            return values.stream()
+                .filter(StringUtils::isNotBlank)
+                .collect(Collectors.joining(StringUtils.SPACE));
+        }
```

Comment:
I feel these could be reused by other mappers. We could extract these out into a utility class?

Reply:
I moved it to TextUtlis